### ISSUE TITLE
Support partitioning parameters when writing data to BigQuery

### ISSUE THREAD

**Issue by kmjung** [GoogleCloudDataproc/spark-bigquery-connector#134]
I want to do e.g.:

```scala
df
  .orderBy($"field1", $"field2")
  .write
  .partitionBy("field1", "field2")
  .format("bigquery")
  .mode("overwrite")
  .option("table", "project:dataset.table")
  .save()
```

I can't specify partitioning options today.

**Comment by tfayyaz** [comment]
It would also be useful to able to use .clusterBy 

eg.

```
df
  .orderBy($"field1", $"field2")
  .write
  .partitionBy("field1", "field2")
  .clusterBy("field1", "field2", "field3", "field4")
  .format("bigquery")
  .mode("overwrite")
  .option("table", "project:dataset.table")
  .save()
```

**Comment by davidrabinowitz** [comment]
Fixed by PR #122
